### PR TITLE
Add globus-compute-sdk and globus-compute-common

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/globus-compute-common/LICENSE
Comment (dopplershift) [inline recipes/globus-compute-common/LICENSE:1]: Where did this come from? I don't see it anywhere upstream?

Comment (jan-janssen) [reply recipes/globus-compute-common/LICENSE:1]: The `setup.cfg` mentions the `Apache Software License` https://github.com/funcx-faas/funcx-common/blob/main/setup.cfg#L14 
So I took the license from their main repository https://github.com/funcx-faas/funcX/blob/main/LICENSE In addition I opened a pull request to include the License upstream https://github.com/funcx-faas/funcx-common/pull/84

Comment (dopplershift) [reply recipes/globus-compute-common/LICENSE:1]: Unfortunately, that's not enough to confirm we have a legal right to distribute that code. For me to merge, I need to see it in either the PyPI tarball or in the upstream repository.

Comment (ocefpaf) [reply recipes/globus-compute-common/LICENSE:1]: BTW, Apache-2.0 license requires the header but not the text like this. In fact they encourage folks to just use the link to the license in the header. See https://www.apache.org/licenses/LICENSE-2.0#apply

**Pull request closed without merging**